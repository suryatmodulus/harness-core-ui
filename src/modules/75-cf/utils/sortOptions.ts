/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export interface Option {
  label: string
  value: string
}

export const sortOptions = (options: Option[]): Option[] =>
  options.sort(({ label: a }, { label: b }) => a.localeCompare(b))
