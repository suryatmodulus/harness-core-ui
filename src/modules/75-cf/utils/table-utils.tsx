/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import type { CellProps } from 'react-table'

export function withTableData<Data extends Record<any, any>, T>(
  mapDataToProps: (data: CellProps<Data>) => T
): (Comp: React.FC<T>) => React.FC<CellProps<Data>> {
  return (Comp: React.FC<T>) => (cellProps: CellProps<Data>) => <Comp {...mapDataToProps(cellProps)} />
}
