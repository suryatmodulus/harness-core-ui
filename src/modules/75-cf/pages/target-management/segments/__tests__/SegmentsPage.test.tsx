/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { fireEvent, getAllByText, getByText, render, waitFor } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import mockEnvironments from '@cf/pages/environments/__tests__/mockEnvironments'
import mockImport from 'framework/utils/mockImport'
import { SegmentsPage } from '../SegmentsPage'

// eslint-disable-next-line jest/no-disabled-tests
describe('SegmentsPage', () => {
  test('SegmentsPage should render loading correctly 1', async () => {
    mockImport('services/cf', {
      useGetAllSegments: () => ({ loading: true, refetch: jest.fn() })
    })

    mockImport('@cf/hooks/useEnvironmentSelectV2', {
      useEnvironmentSelectV2: () => ({
        data: undefined,
        loading: true,
        error: undefined,
        refetch: jest.fn(),
        EnvironmentSelect: function EnvironmentSelect() {
          return <div />
        }
      })
    })

    const { container } = render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentsPage />
      </TestWrapper>
    )

    expect(container.querySelector('[data-icon="steps-spinner"]')).toBeDefined()
  })

  test('SegmentsPage should render loading correctly 2', async () => {
    mockImport('services/cf', {
      useGetAllSegments: () => ({ loading: false, data: { segments: [] }, refetch: jest.fn() })
    })

    mockImport('@cf/hooks/useEnvironmentSelectV2', {
      useEnvironmentSelectV2: () => ({
        data: undefined,
        loading: true,
        error: undefined,
        refetch: jest.fn(),
        EnvironmentSelect: function EnvironmentSelect() {
          return <div />
        }
      })
    })

    const { container } = render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentsPage />
      </TestWrapper>
    )

    expect(container.querySelector('[data-icon="steps-spinner"]')).toBeDefined()
  })

  test('SegmentsPage should render loading correctly 3', async () => {
    mockImport('services/cf', {
      useGetAllSegments: () => ({ loading: true, refetch: jest.fn() })
    })

    mockImport('@cf/hooks/useEnvironmentSelectV2', {
      useEnvironmentSelectV2: () => ({
        environments: [],
        loading: false,
        error: undefined,
        refetch: jest.fn(),
        EnvironmentSelect: function EnvironmentSelect() {
          return <div />
        }
      })
    })

    const { container } = render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentsPage />
      </TestWrapper>
    )

    expect(container.querySelector('[data-icon="steps-spinner"]')).toBeDefined()
  })

  test('SegmentsPage should render error correctly 1', async () => {
    const error = { message: 'SOME ERROR OCCURS' }
    mockImport('services/cf', {
      useGetAllSegments: () => ({ error, loading: false, refetch: jest.fn() })
    })

    mockImport('@cf/hooks/useEnvironmentSelectV2', {
      useEnvironmentSelectV2: () => ({
        environments: [{}],
        loading: false,
        error: undefined,
        refetch: jest.fn(),
        EnvironmentSelect: function EnvironmentSelect() {
          return <div />
        }
      })
    })

    render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentsPage />
      </TestWrapper>
    )

    expect(getByText(document.body, error.message)).toBeDefined()
  })

  test('SegmentsPage should render error correctly 2', async () => {
    const error = { message: 'SOME ERROR OCCURS' }
    mockImport('services/cf', {
      useGetAllSegments: () => ({ data: [], loading: false, refetch: jest.fn() })
    })

    mockImport('@cf/hooks/useEnvironmentSelectV2', {
      useEnvironmentSelectV2: () => ({
        environments: undefined,
        loading: false,
        error,
        refetch: jest.fn(),
        EnvironmentSelect: function EnvironmentSelect() {
          return <div />
        }
      })
    })

    render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentsPage />
      </TestWrapper>
    )

    expect(getByText(document.body, error.message)).toBeDefined()
  })

  test('SegmentsPage should render data correctly', async () => {
    mockImport('services/cf', {
      useGetAllSegments: () => ({
        data: {
          segments: [
            {
              createdAt: 1619397730112,
              environment: 'QA',
              excluded: [],
              identifier: 'segment1',
              included: [],
              name: 'segment1',
              rules: []
            }
          ]
        },
        loading: false,
        refetch: jest.fn()
      })
    })

    mockImport('@cf/hooks/useEnvironmentSelectV2', {
      useEnvironmentSelectV2: () => ({
        environments: mockEnvironments,
        loading: false,
        refetch: jest.fn(),
        EnvironmentSelect: function EnvironmentSelect() {
          return <div />
        }
      })
    })

    const { container } = render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentsPage />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('No data view should be rendered', async () => {
    mockImport('services/cf', {
      useGetAllSegments: () => ({
        data: { segments: [] },
        loading: false,
        refetch: jest.fn()
      })
    })

    mockImport('@cf/hooks/useEnvironmentSelectV2', {
      useEnvironmentSelectV2: () => ({
        environments: mockEnvironments,
        loading: false,
        refetch: jest.fn(),
        EnvironmentSelect: function EnvironmentSelect() {
          return <div />
        }
      })
    })

    const { container } = render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentsPage />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('No environment view should be rendered', async () => {
    mockImport('services/cf', {
      useGetAllSegments: () => ({
        data: { segments: [] },
        loading: false,
        refetch: jest.fn()
      })
    })

    mockImport('@cf/hooks/useEnvironmentSelectV2', {
      useEnvironmentSelectV2: () => ({
        environments: [],
        loading: false,
        refetch: jest.fn(),
        EnvironmentSelect: function EnvironmentSelect() {
          return <div />
        }
      })
    })

    const { container } = render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentsPage />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('No environment view should be rendered', async () => {
    mockImport('services/cf', {
      useGetAllSegments: () => ({
        data: { segments: [] },
        loading: false,
        refetch: jest.fn()
      })
    })

    mockImport('@cf/hooks/useEnvironmentSelectV2', {
      useEnvironmentSelectV2: () => ({
        environments: [],
        loading: false,
        refetch: jest.fn(),
        EnvironmentSelect: function EnvironmentSelect() {
          return <div />
        }
      })
    })

    const { container } = render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentsPage />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('Should go to edit page by clicking a row', async () => {
    mockImport('services/cf', {
      useGetAllSegments: () => ({
        data: {
          segments: [
            {
              createdAt: 1619397730112,
              environment: 'QA',
              excluded: [],
              identifier: 'segment1',
              included: [],
              name: 'segment1',
              rules: []
            }
          ]
        },
        loading: false,
        refetch: jest.fn()
      })
    })

    mockImport('@cf/hooks/useEnvironmentSelectV2', {
      useEnvironmentSelectV2: () => ({
        environments: mockEnvironments,
        loading: false,
        refetch: jest.fn(),
        EnvironmentSelect: function EnvironmentSelect() {
          return <div />
        }
      })
    })

    const { container } = render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/target-management/target-groups"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentsPage />
      </TestWrapper>
    )

    await waitFor(() => expect(getAllByText(container, 'segment1')).toBeTruthy())

    fireEvent.click(container.getElementsByClassName('TableV2--row TableV2--card TableV2--clickable')[0] as HTMLElement)

    expect(
      getByText(container, '/account/dummy/cf/orgs/dummy/projects/dummy/target-management/target-groups/segment1')
    ).toBeDefined()
  })

  test('Should go to edit page by clicking edit', async () => {
    mockImport('services/cf', {
      useGetAllSegments: () => ({
        data: {
          segments: [
            {
              createdAt: 1619397730112,
              environment: 'QA',
              excluded: [],
              identifier: 'segment1',
              included: [],
              name: 'segment1',
              rules: []
            }
          ]
        },
        loading: false,
        refetch: jest.fn()
      })
    })

    mockImport('@cf/hooks/useEnvironmentSelectV2', {
      useEnvironmentSelectV2: () => ({
        environments: mockEnvironments,
        loading: false,
        refetch: jest.fn(),
        EnvironmentSelect: function EnvironmentSelect() {
          return <div />
        }
      })
    })

    const { container } = render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/target-management/target-groups"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentsPage />
      </TestWrapper>
    )

    await waitFor(() => expect(getAllByText(container, 'segment1')).toBeTruthy())

    fireEvent.click(container.querySelector('[data-icon="Options"]') as HTMLElement)
    fireEvent.click(document.querySelector('[icon="edit"]') as HTMLElement)

    expect(
      getByText(container, '/account/dummy/cf/orgs/dummy/projects/dummy/target-management/target-groups/segment1')
    ).toBeDefined()
  })

  test('Should allow deleting', async () => {
    const mutate = jest.fn(() => {
      return Promise.resolve({ data: {} })
    })

    mockImport('services/cf', {
      useGetAllSegments: () => ({
        data: {
          segments: [
            {
              createdAt: 1619397730112,
              environment: 'QA',
              excluded: [],
              identifier: 'segment1',
              included: [],
              name: 'segment1',
              rules: []
            }
          ]
        },
        loading: false,
        refetch: jest.fn()
      }),
      useDeleteSegment: () => ({ mutate })
    })

    mockImport('@cf/hooks/useEnvironmentSelectV2', {
      useEnvironmentSelectV2: () => ({
        environments: mockEnvironments,
        loading: false,
        refetch: jest.fn(),
        EnvironmentSelect: function EnvironmentSelect() {
          return <div />
        }
      })
    })

    const { container } = render(
      <TestWrapper
        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/target-management/target-groups"
        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
      >
        <SegmentsPage />
      </TestWrapper>
    )

    await waitFor(() => expect(getAllByText(container, 'segment1')).toBeTruthy())

    fireEvent.click(container.querySelector('[data-icon="Options"]') as HTMLElement)
    fireEvent.click(document.querySelector('[icon="trash"]') as HTMLElement)

    fireEvent.click(document.querySelector('button[class*=intent-danger]') as HTMLButtonElement)
    await waitFor(() => expect(mutate).toBeCalledTimes(1))
  })
})
