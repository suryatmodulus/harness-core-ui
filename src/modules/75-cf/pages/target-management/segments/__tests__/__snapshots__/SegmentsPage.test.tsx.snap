// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentsPage No data view should be rendered 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
    data-name="header"
    style="border-bottom: 0.5px solid #d9dae6; display: flex; height: 80px;"
  >
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--padding-right-xlarge"
      style="flex-grow: 1;"
    >
      <h1
        class="StyledProps--font StyledProps--font-h1 StyledProps--main"
        data-name="header"
        level="1"
        style="font-weight: bold; font-size: 20px; line-height: 28px; color: rgb(34, 39, 45); display: flex; align-items: center;"
      >
        cf.shared.targetManagement
      </h1>
      <span
        style="flex-grow: 1;"
      />
      <ul
        class="wrapper"
        data-testid="CFSectionToggle"
      >
        <li
          class="item"
        >
          <a
            class="link"
            href="/account/dummy/cf/orgs/dummy/projects/dummy/target-management/targets?activeEnvironment=undefined"
          >
            <span
              class="align"
              data-tooltip-id="ff_targets_heading"
            >
              cf.shared.targets
            </span>
          </a>
        </li>
        <li
          class="item"
        >
          <a
            class="link"
            href="/account/dummy/cf/orgs/dummy/projects/dummy/target-management/target-groups?activeEnvironment=undefined"
          >
            <span
              class="align"
              data-tooltip-id="ff_segments_heading"
            >
              cf.shared.segments
            </span>
          </a>
        </li>
      </ul>
      <span
        style="flex-grow: 1;"
      />
      <div
        data-testid="CFTargetManagementHeaderSpacer"
        style="width: 325px;"
      />
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main"
    data-name="body"
    style="height: calc(100% - 80px); overflow: hidden; background-color: rgb(248, 250, 251);"
  >
    <div
      class="StyledProps--font StyledProps--main"
      data-name="toolbar"
      style="height: 64px; border-bottom: 0.5px solid #d9dae6; align-items: center; top: 0px; z-index: 1;"
    >
      <div
        class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
      >
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            cf.segments.create
          </span>
        </button>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-small StyledProps--color StyledProps--color-grey400"
          style="align-self: center;"
        >
          cf.segments.pageDescription
        </p>
        <span
          style="flex-grow: 1;"
        />
        <div
          class="bp3-input-group ui-search-box ExpandingSearchInput--main    ExpandingSearchInput--light"
          data-name="findFlag"
        >
          <span
            class="bp3-icon StyledProps--main ExpandingSearchInput--icon StyledProps--color StyledProps--color-grey700"
            data-icon="thinner-search"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 16 16"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
          <input
            class="bp3-input  ExpandingSearchInput--filter"
            dir="auto"
            placeholder="search"
            style="padding-right: 0px;"
            type="search"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main"
      data-name="content"
      style="height: calc(100% - 64px - 0px); overflow: auto;"
    >
      <div
        class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-align-center-center"
        style="width: 100%; height: 100%;"
      >
        <div
          class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
          style="align-items: center; margin-top: -48px; width: 470px;"
        >
          <img
            height="220"
            src="test-file-stub"
            width="220"
          />
          <h2
            class="StyledProps--font StyledProps--font-h2 StyledProps--main"
            level="2"
            style="font-weight: 600; font-size: 20px; line-height: 28px; color: rgb(137, 137, 143); text-align: center; padding: 30px 0px;"
          >
            cf.segments.noSegment
          </h2>
          <button
            class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              cf.segments.create
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SegmentsPage No environment view should be rendered 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
    data-name="header"
    style="border-bottom: 0.5px solid #d9dae6; display: flex; height: 80px;"
  >
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--padding-right-xlarge"
      style="flex-grow: 1;"
    >
      <h1
        class="StyledProps--font StyledProps--font-h1 StyledProps--main"
        data-name="header"
        level="1"
        style="font-weight: bold; font-size: 20px; line-height: 28px; color: rgb(34, 39, 45); display: flex; align-items: center;"
      >
        cf.shared.targetManagement
      </h1>
      <span
        style="flex-grow: 1;"
      />
      <ul
        class="wrapper"
        data-testid="CFSectionToggle"
      >
        <li
          class="item"
        >
          <a
            class="link"
            href="/account/dummy/cf/orgs/dummy/projects/dummy/target-management/targets?activeEnvironment=undefined"
          >
            <span
              class="align"
              data-tooltip-id="ff_targets_heading"
            >
              cf.shared.targets
            </span>
          </a>
        </li>
        <li
          class="item"
        >
          <a
            class="link"
            href="/account/dummy/cf/orgs/dummy/projects/dummy/target-management/target-groups?activeEnvironment=undefined"
          >
            <span
              class="align"
              data-tooltip-id="ff_segments_heading"
            >
              cf.shared.segments
            </span>
          </a>
        </li>
      </ul>
      <span
        style="flex-grow: 1;"
      />
      <div
        data-testid="CFTargetManagementHeaderSpacer"
        style="width: 325px;"
      />
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main"
    data-name="body"
    style="height: calc(100% - 80px); overflow: hidden; background-color: rgb(248, 250, 251);"
  >
    <div
      class="StyledProps--font StyledProps--main"
      data-name="content"
      style="height: calc(100% - 0px - 0px); overflow: auto;"
    >
      <div
        class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-align-center-center"
        style="height: 100%;"
      >
        <div
          class="StyledProps--font StyledProps--main StyledProps--flex"
          style="flex-direction: column;"
        >
          <img
            height="220"
            src="test-file-stub"
            width="220"
          />
          <h2
            class="StyledProps--font StyledProps--font-h2 StyledProps--main"
            level="2"
            style="padding-top: 50px; font-weight: 600; font-size: 20px; line-height: 28px; color: rgb(34, 34, 42);"
          >
            cf.noEnvironment.title
          </h2>
          <p
            class="StyledProps--font StyledProps--main"
            style="line-height: 18px; color: rgb(34, 34, 42); text-align: center; width: 400px;"
          >
            cf.noEnvironment.message
          </p>
          <button
            class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--padding-top-small StyledProps--padding-bottom-small StyledProps--padding-left-huge StyledProps--padding-right-huge Button--with-current-color"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              cf.environments.create.title
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SegmentsPage No environment view should be rendered 2`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
    data-name="header"
    style="border-bottom: 0.5px solid #d9dae6; display: flex; height: 80px;"
  >
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--padding-right-xlarge"
      style="flex-grow: 1;"
    >
      <h1
        class="StyledProps--font StyledProps--font-h1 StyledProps--main"
        data-name="header"
        level="1"
        style="font-weight: bold; font-size: 20px; line-height: 28px; color: rgb(34, 39, 45); display: flex; align-items: center;"
      >
        cf.shared.targetManagement
      </h1>
      <span
        style="flex-grow: 1;"
      />
      <ul
        class="wrapper"
        data-testid="CFSectionToggle"
      >
        <li
          class="item"
        >
          <a
            class="link"
            href="/account/dummy/cf/orgs/dummy/projects/dummy/target-management/targets?activeEnvironment=undefined"
          >
            <span
              class="align"
              data-tooltip-id="ff_targets_heading"
            >
              cf.shared.targets
            </span>
          </a>
        </li>
        <li
          class="item"
        >
          <a
            class="link"
            href="/account/dummy/cf/orgs/dummy/projects/dummy/target-management/target-groups?activeEnvironment=undefined"
          >
            <span
              class="align"
              data-tooltip-id="ff_segments_heading"
            >
              cf.shared.segments
            </span>
          </a>
        </li>
      </ul>
      <span
        style="flex-grow: 1;"
      />
      <div
        data-testid="CFTargetManagementHeaderSpacer"
        style="width: 325px;"
      />
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main"
    data-name="body"
    style="height: calc(100% - 80px); overflow: hidden; background-color: rgb(248, 250, 251);"
  >
    <div
      class="StyledProps--font StyledProps--main"
      data-name="content"
      style="height: calc(100% - 0px - 0px); overflow: auto;"
    >
      <div
        class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-align-center-center"
        style="height: 100%;"
      >
        <div
          class="StyledProps--font StyledProps--main StyledProps--flex"
          style="flex-direction: column;"
        >
          <img
            height="220"
            src="test-file-stub"
            width="220"
          />
          <h2
            class="StyledProps--font StyledProps--font-h2 StyledProps--main"
            level="2"
            style="padding-top: 50px; font-weight: 600; font-size: 20px; line-height: 28px; color: rgb(34, 34, 42);"
          >
            cf.noEnvironment.title
          </h2>
          <p
            class="StyledProps--font StyledProps--main"
            style="line-height: 18px; color: rgb(34, 34, 42); text-align: center; width: 400px;"
          >
            cf.noEnvironment.message
          </p>
          <button
            class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--padding-top-small StyledProps--padding-bottom-small StyledProps--padding-left-huge StyledProps--padding-right-huge Button--with-current-color"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              cf.environments.create.title
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SegmentsPage SegmentsPage should render data correctly 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
    data-name="header"
    style="border-bottom: 0.5px solid #d9dae6; display: flex; height: 80px;"
  >
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--padding-right-xlarge"
      style="flex-grow: 1;"
    >
      <h1
        class="StyledProps--font StyledProps--font-h1 StyledProps--main"
        data-name="header"
        level="1"
        style="font-weight: bold; font-size: 20px; line-height: 28px; color: rgb(34, 39, 45); display: flex; align-items: center;"
      >
        cf.shared.targetManagement
      </h1>
      <span
        style="flex-grow: 1;"
      />
      <ul
        class="wrapper"
        data-testid="CFSectionToggle"
      >
        <li
          class="item"
        >
          <a
            class="link"
            href="/account/dummy/cf/orgs/dummy/projects/dummy/target-management/targets?activeEnvironment=undefined"
          >
            <span
              class="align"
              data-tooltip-id="ff_targets_heading"
            >
              cf.shared.targets
            </span>
          </a>
        </li>
        <li
          class="item"
        >
          <a
            class="link"
            href="/account/dummy/cf/orgs/dummy/projects/dummy/target-management/target-groups?activeEnvironment=undefined"
          >
            <span
              class="align"
              data-tooltip-id="ff_segments_heading"
            >
              cf.shared.segments
            </span>
          </a>
        </li>
      </ul>
      <span
        style="flex-grow: 1;"
      />
      <div
        data-testid="CFTargetManagementHeaderSpacer"
        style="width: 325px;"
      />
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main"
    data-name="body"
    style="height: calc(100% - 80px); overflow: hidden; background-color: rgb(248, 250, 251);"
  >
    <div
      class="StyledProps--font StyledProps--main"
      data-name="toolbar"
      style="height: 64px; border-bottom: 0.5px solid #d9dae6; align-items: center; top: 0px; z-index: 1;"
    >
      <div
        class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
      >
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            cf.segments.create
          </span>
        </button>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-size-small StyledProps--color StyledProps--color-grey400"
          style="align-self: center;"
        >
          cf.segments.pageDescription
        </p>
        <span
          style="flex-grow: 1;"
        />
        <div
          class="bp3-input-group ui-search-box ExpandingSearchInput--main    ExpandingSearchInput--light"
          data-name="findFlag"
        >
          <span
            class="bp3-icon StyledProps--main ExpandingSearchInput--icon StyledProps--color StyledProps--color-grey700"
            data-icon="thinner-search"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 16 16"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
          <input
            class="bp3-input  ExpandingSearchInput--filter"
            dir="auto"
            placeholder="search"
            style="padding-right: 0px;"
            type="search"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main"
      data-name="content"
      style="height: calc(100% - 64px - 60px); overflow: auto;"
    >
      <div
        class="StyledProps--font StyledProps--main StyledProps--padding-top-medium StyledProps--padding-right-xxlarge StyledProps--padding-left-xxlarge"
      >
        <div
          class="TableV2--table"
        >
          <div
            class="TableV2--header"
            role="row"
          >
            <div
              class="TableV2--cell"
              colspan="1"
              role="columnheader"
              style="width: 35%;"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
              >
                CF.SHARED.SEGMENT
              </p>
            </div>
            <div
              class="TableV2--cell"
              colspan="1"
              role="columnheader"
              style="width: 35%;"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
              >
                IDENTIFIER
              </p>
            </div>
            <div
              class="TableV2--cell"
              colspan="1"
              role="columnheader"
              style="width: 30%;"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
              >
                CF.TARGETS.CREATEDDATE
              </p>
            </div>
          </div>
          <div
            class="TableV2--body"
          >
            <div
              class="TableV2--row TableV2--card TableV2--clickable"
              role="row"
            >
              <div
                class="TableV2--cell"
                role="cell"
                style="width: 35%;"
              >
                <div
                  class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main"
                  style="align-items: center;"
                >
                  <div
                    class="StyledProps--font StyledProps--main StyledProps--margin-right-small"
                  >
                    <ul
                      class="stack"
                      style="--size: 30px; --itemContainerSize: 30px;"
                    >
                      <li
                        class="itemContainer"
                        style="--color: var(--white); --background: #BDA5F2;"
                      >
                        <p
                          class="StyledProps--font StyledProps--main"
                        >
                          SE
                        </p>
                      </li>
                    </ul>
                  </div>
                  <div
                    class="StyledProps--font StyledProps--main"
                  >
                    <p
                      class="StyledProps--font StyledProps--main"
                      style="font-weight: 600; line-height: 24px; color: rgb(34, 34, 42);"
                    >
                      segment1
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="TableV2--cell"
                role="cell"
                style="width: 35%;"
              >
                <p
                  class="StyledProps--font StyledProps--main"
                >
                  segment1
                </p>
              </div>
              <div
                class="TableV2--cell"
                role="cell"
                style="width: 30%;"
              >
                <div
                  class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-distribution-space-between StyledProps--flex-align-center-center"
                >
                  <p
                    class="StyledProps--font StyledProps--main"
                  >
                    dummy date
                  </p>
                  <div
                    class="StyledProps--font StyledProps--main"
                    style="text-align: right;"
                  >
                    <span
                      class="bp3-popover-wrapper"
                    >
                      <span
                        class="bp3-popover-target"
                      >
                        <button
                          class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
                          type="button"
                        >
                          <span
                            class="bp3-icon StyledProps--main"
                            data-icon="Options"
                          >
                            <svg
                              fill="none"
                              height="16"
                              viewBox="0 0 4 16"
                              width="16"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                clip-rule="evenodd"
                                d="M2 12.5a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zm2-12a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zM2 7a1 1 0 100 2 1 1 0 000-2zM0 8a2 2 0 114 0 2 2 0 01-4 0z"
                                fill="currentColor"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </button>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main pagination"
      data-name="pagination"
      style="height: 60px;"
    />
  </div>
</div>
`;
